internal/bundle/file.go

```diff
@@ -0,0 +1,59 @@
+package bundle
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+const (
+	markerFileName = ".bundle"
+)
+
+var (
+	execDir string
+)
+
+func executableDirectory() string {
+	if execDir != "" {
+		return execDir
+	}
+
+	// Get the absolute path of the executable
+	executablePath, err := os.Executable()
+	if err != nil {
+		panic(fmt.Sprintf("Failed determining executable path: %v\n", err))
+	}
+
+	// Resolve the absolute path (eliminate any symbolic links)
+	absolutePath, err := filepath.EvalSymlinks(executablePath)
+	if err != nil {
+		panic(fmt.Sprintf("Failed determining executable path: %v\n", err))
+	}
+
+	execDir = filepath.Dir(absolutePath)
+	return execDir
+}
+
+func IsInsideBundle() bool {
```

Comment:
I think if we're going support both tar and path options, it's better to use current directory instead of binary location. What if you copied `gohomecli` to `/usr/local/bin` and running update from there? 
Are you're expecting having `/usr/local/bin/gohomecli.bundle` as well?

BTW thx for `os.Stat`, I totally forgot about it

Reply:
I would take this:
If directory is provided - use provided directory, otherwise use current directory